### PR TITLE
React: Hookup datasets page using react-query

### PR DESCRIPTION
**Pull request by frewmack**
Part of #284

### CODE REVIEW

File: react/src/Datasets/components/DatasetInfoDialog.tsx [hunk 1]
```diff
@@ -104,7 +92,7 @@ export default function DatasetInfoDialog(props: DialogProp) {
                 </div>
                 <Divider />
                 <div className={classes.infoSection}>
-                    {analyses.length > 0 && (
+                    {analyses && analyses.length > 0 && (
```
Comment (kevinlul) [inline react/src/Datasets/components/DatasetInfoDialog.tsx:95]:
```suggestion
                    {analyses?.length > 0 && (
```

File: react/src/Participants/components/ParticipantInfoDialog.tsx [hunk 2]
```diff
@@ -49,37 +48,22 @@ interface DialogProp {
...
 export default function ParticipantInfoDialog(props: DialogProp) {
     const classes = useStyles();
+    const datasets = useMemo(
+        () => props.participant.tissue_samples.flatMap(sample => sample.datasets),
+        [props.participant]
+    );
+    const datasetResults = useDatasetQueries(datasets.map(d => d.dataset_id));
+    const analyses = useMemo(
+        () =>
+            datasetResults.reduce(
```
Comment (kevinlul) [inline react/src/Participants/components/ParticipantInfoDialog.tsx:58]:
```suggestion
            datasetResults.reduce<Analysis[]>(
```

File: react/src/Participants/components/ParticipantInfoDialog.tsx [hunk 3]
```diff
@@ -49,37 +48,22 @@ interface DialogProp {
...
+                (prev, curr) => (curr.isSuccess ? prev.concat(curr.data.analyses) : prev),
+                [] as Analysis[]
```
Comment (kevinlul) [inline react/src/Participants/components/ParticipantInfoDialog.tsx:60]:
```suggestion
                []
```

File: react/src/hooks/datasets/useDatasetsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Used with material-table's remote data feature.
+ */
+export function useDatasetsPage() {
+    const queryClient = useQueryClient();
+
+    return async (query: Query<Dataset>) => {
+        return await queryClient.fetchQuery<QueryResult<Dataset>, Error>(
+            ["datasets", query],
+            () => fetchDatasets(query) as Promise<QueryResult<Dataset>>
```
Comment (kevinlul) [inline react/src/hooks/datasets/useDatasetsPage.tsx:23]: Is this cast necessary?

File: react/src/Datasets/components/DatasetInfoDialog.tsx [hunk 5]
```diff
@@ -41,30 +41,18 @@ interface DialogProp {
     open: boolean;
     dataset: Dataset;
     onClose: () => void;
-    onUpdate: (dataset_id: string, newDataset: { [key: string]: any }) => void;
+    onUpdate?: (dataset_id: string, newDataset: { [key: string]: any }) => void;
```
Comment (kevinlul) [inline react/src/Datasets/components/DatasetInfoDialog.tsx:44]: Is this still used?

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: react/src/Datasets/components/DatasetInfoDialog.tsx (hunk 1 above)
Comment (frewmack) [reply react/src/Datasets/components/DatasetInfoDialog.tsx:95]: This change doesn't work here. `Object is possibly undefined`

Comment (kevinlul) [reply react/src/Datasets/components/DatasetInfoDialog.tsx:95]: That's a bit strange alright. Is this a side effect of useMemo?

Comment (frewmack) [reply react/src/Datasets/components/DatasetInfoDialog.tsx:95]: I don't think so. The same error occurs when I try `dataset?.analyses?.length` as well, avoiding useMemo.

**Review by kevinlul — APPROVED** [review]
LGTM!

**Pull request merged**